source/CubeTexture.ts

```diff
@@ -0,0 +1,336 @@
+
+import { assert } from './auxiliaries';
+import { byteSizeOfFormat } from './formatbytesizes';
+
+import { Bindable } from './bindable';
+import { Initializable } from './initializable';
+import { AbstractObject } from './object';
+
+
+type AnyImageData = ArrayBufferView | ImageData | HTMLImageElement | HTMLCanvasElement | HTMLVideoElement |
+    ImageBitmap | undefined;
+
+/**
+ * Wrapper for an WebGL 2D texture providing size accessors and requiring for bind, unbind, resize, validity, and
+ * initialization implementations. The texture object is created on initialization and deleted on uninitialization.
+ * ```
+ * @todo add usage example
+ * ```
+ */
+export class CubeTexture extends AbstractObject<WebGLTexture> implements Bindable {
+
+    /**
+     * Default texture, e.g., used for unbind.
+     */
+    static readonly DEFAULT_TEXTURE = undefined;
+
+    /** @see {@link width} */
+    protected _width: GLsizei = 0;
+
+    /** @see {@link height} */
+    protected _height: GLsizei = 0;
+
+    /** @see {@link internalFormat} */
+    protected _internalFormat: GLenum | undefined = undefined;
+
+    /** @see {@link format} */
+    protected _format: GLenum | undefined = undefined;
+
+    /** @see {@link type} */
+    protected _type: GLenum | undefined = undefined;
+
+    /**
+     * Create a texture object on the GPU.
+     * @param width - Initial width of the texture in px.
+     * @param height - Initial height of the texture in px.
+     * @param internalFormat - Internal format of the texture object.
+     * @param format - Format of the texture data even though no data is passed.
+     * @param type - Data type of the texel data.
+     */
+    protected create(width: GLsizei, height: GLsizei, internalFormat: GLenum,
+        format: GLenum, type: GLenum): WebGLTexture | undefined {
+
+        assert(width > 0 && height > 0, `texture requires valid width and height of greater than zero`);
+        const gl = this._context.gl;
+
+        this._object = gl.createTexture();
+
+        this._width = width;
+        this._height = height;
+        this._internalFormat = internalFormat;
+        this._format = format;
+        this._type = type;
+
+        gl.bindTexture(gl.GL_TEXTURE_CUBE_MAP, this._object);
+
+        gl.texParameter(gl.TEXTURE_CUBE_MAP, gl.TEXTURE_MIN_FILTER, gl.LINEAR);
+        gl.texParameter(gl.TEXTURE_CUBE_MAP, gl.TEXTURE_MAG_FILTER, gl.LINEAR);
+        gl.texParameter(gl.TEXTURE_CUBE_MAP, gl.TEXTURE_WRAP_S, gl.CLAMP_TO_EDGE);
+        gl.texParameter(gl.TEXTURE_CUBE_MAP, gl.TEXTURE_WRAP_T, gl.CLAMP_TO_EDGE);
+
+        gl.texImage2D(gl.TEXTURE_CUBE_MAP_POSITIVE_X, 0, this._internalFormat, this._width, this._height, 0,
```

Comment:
minor: please adhere to the following sequence order:
* `TEXTURE_CUBE_MAP_POSITIVE_X`
* `TEXTURE_CUBE_MAP_NEGATIVE_X`
* `TEXTURE_CUBE_MAP_POSITIVE_Y`
* `TEXTURE_CUBE_MAP_NEGATIVE_Y`
* `TEXTURE_CUBE_MAP_POSITIVE_Z`
* `TEXTURE_CUBE_MAP_NEGATIVE_Z`

Since this is the most common sequence and the enums are typically specified in that order as well.